src/eval/array.rs

```diff
@@ -199,6 +200,36 @@ impl Array {
         Ok(acc)
     }
 
+    /// Calculates the sum of the array's items
+    pub fn sum(&self, default: Option<Value>, span: Span) -> SourceResult<Value> {
+        let mut acc = self
+            .first()
+            .map(|x| x.clone())
+            .or(default.ok_or_else(|| {
+                eco_format!("cannot calculate sum of empty array with no default")
+            }))
+            .at(span)?;
+        for i in self.iter().skip(1) {
+            acc = add(acc, i.clone()).at(span)?;
+        }
+        Ok(acc)
+    }
+
+    /// Calculates the product of the array's items
+    pub fn product(&self, default: Option<Value>, span: Span) -> SourceResult<Value> {
+        let mut acc = self
+            .first()
+            .map(|x| x.clone())
+            .or(default.ok_or_else(|| {
+                eco_format!("cannot calculate product of empty array with no default")
+            }))
```

Comment:
Same comment as above:
```suggestion
            .or(default)
            .ok_or_else(|| {
                eco_format!("cannot calculate product of empty array with no default")
            })
```